### PR TITLE
fix ratio modifier application

### PR DESCRIPTION
**Pull request by jnq-t**
we were applying the ratio to the input frequency of the scale degree rather than to the starting frequency of the whole scale. applied a fix. @pluto-bell  i can't figure out how to test now that you added the FE stuff 🤣  maybe we could test this tgth before i merge it?

### CODE REVIEW

File: app/TEMP_classes_workspace.js [hunk 1]
```diff
@@ -66,17 +66,12 @@ class ScaleDegree {
         this.frequency = this.#frequency()
     }
     #frequency() {
-        let frequency = this.inputFrequency;
-        const ratio = this.#ratio();
-        frequency *= ratio; // if no ratio is provided this will be 1
-        frequency *= Math.pow(2, this.modifier.detuneByCents/1200); // if no detune is provided this will be 1
-        return frequency;
+        if (this.modifier.ratioNumerator > 0 && this.modifier.ratioDenominator > 0) return this.#ratio(this.startingFreq);
+        return this.inputFrequency *= Math.pow(2, this.modifier.detuneByCents/1200); // if no detune is provided this will be 1
```
Comment (pluto-bell) [inline app/TEMP_classes_workspace.js]: make helper method for detune by cents

File: app/TEMP_classes_workspace.js [hunk 2]
```diff
@@ -66,17 +66,12 @@ class ScaleDegree {
         this.frequency = this.#frequency()
     }
     #frequency() {
-        let frequency = this.inputFrequency;
-        const ratio = this.#ratio();
-        frequency *= ratio; // if no ratio is provided this will be 1
-        frequency *= Math.pow(2, this.modifier.detuneByCents/1200); // if no detune is provided this will be 1
-        return frequency;
+        if (this.modifier.ratioNumerator > 0 && this.modifier.ratioDenominator > 0) return this.#ratio(this.startingFreq);
```
Comment (pluto-bell) [inline app/TEMP_classes_workspace.js]: re-write so both ratio & detune by cents can be used

**Review by pluto-bell — CHANGES_REQUESTED** [review]
Please review my comments (:

File: app/TEMP_classes_workspace.js (hunk 1 above)
Comment (jnq-t) [reply app/TEMP_classes_workspace.js]: got it!

File: app/TEMP_classes_workspace.js (hunk 2 above)
Comment (jnq-t) [reply app/TEMP_classes_workspace.js]: adding this to the backlog

**Comment by pluto-bell** [comment]
### All freq.'s getting altered

**Documents on**
`(78) [242.7886656724614, 257.22563100871037, 272.5210629770106, 288.72600865970804, 305.89455055662296, 300, 343.3550237844621, 363.7719759624957, 385.40298329442084, 420, 432.6002220531987, 458.32396987700014, 485.5773313449228, 514.4512620174207, 545.0421259540212, 577.4520173194161, 611.7891011132459, 600, 686.7100475689242, 727.5439519249913...]`


**Documents off**
`(77) [240, 254.27114264623089, 269.3908915942495, 285.40970760065306, 302.3810519747696, 320.3615650008083, 339.4112549695429, 359.5936984504036, 380.97625247236795, 403.63027932178306, 427.631384707363, 453.059670087213, 480, 508.54228529246177, 538.781783188499, 570.8194152013061, 604.7621039495392, 640.7231300016166, 678.8225099390858...]`

**Pull request merged**